app/dashboard/views.py

```diff
@@ -3090,6 +3094,100 @@ def verify_user_google(request):
 
     return redirect('profile_by_tab', 'trust')
 
+
+@login_required
+def verify_profile_with_ens(request):
+    profile = request.user.profile
+    user_address = profile.preferred_payout_address
+
+    # 1. Check user has preferred address
+    if not is_address(user_address):
+        return JsonResponse({
+            'error': 'NO_PAYOUT_ADDRESS_ASSOCIATED',
+            'msg': 'You don\'t have one preferred payout address associated to your profile',
+            'data': {
+                'step': 1,
+                'verified': profile.is_ens_verified
+            }
+        })
+
+    # 2. Check Reverse lookup
+    web3 = get_web3('mainnet')
+    ns = ens.ENS.fromWeb3(web3, '0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e')
+    node = ns.name(user_address)
+
+    if not node:
+        return JsonResponse({
+            'error': 'NO_ENS_NAME_ASSOCIATED',
+            'msg': f'You don\'t have one ens domain associated to your address {user_address}',
+            'data': {
+                'step': 2,
+                'verified': profile.is_ens_verified
+            }
+        })
+
+    # 3. Check Forward lookup
+    registered_address = ns.address(node)
+    if not is_address(registered_address):
+        return JsonResponse({
+            'error': 'NO_ADDRESS_ASSOCIATED_TO_ENS',
+            'msg': f'You don\'t have associated your address to your domain {node}',
+            'data': {
+                'step': 3,
+                'verified': profile.is_ens_verified
+            }
+        })
+
+    # 4. Implement challenge
+    resolver_contract = get_ens_resolver_contract('mainnet', node)
+    github_handle = resolver_contract.functions.text(name_to_hash(node), 'vnd.github').call()
+
+    if github_handle != profile.handle:
+        return JsonResponse({
+            'error': 'NO_MATCH_GITHUB_PROFILE',
+            'msg': f'Sorry the {github_handle} github handle registered in ENS {node} doesn\'t match your gitcoin handle ${profile.handle}',
+            'data': {
+                'step': 4,
+                'verified': profile.is_ens_verified
+            }
+        })
+
+    # 5. Check if address matches.
+    if registered_address != user_address:
+        return JsonResponse({
+            'error': 'NO_ADDRESS_DOESNT_MATCH',
+            'msg': f'Sorry the registered ENS address differ from your preferred payout address',
```

Comment:
Please reword to something like `{node} has {registered_address[0:5]}... set as ETH address which is different from your preferred payout address ({user_address[0:5}}...).  Please set your correct ETH address at https://app.ens.domains/name/{node}`

Reply:
sure, i'll do the changes in the morning. Its a bit late here :sleeping:. thank you for your feedback!

Reply:
Thank you so much for working on this. We are trying to make it by the next Gitcoin CLR round 8 which starts in about a week from now. So as long as this PR can get merged and deployed by that time I am fine with it unless @owocki has specific deadline.

Reply:
at this rate i think its mostly about when the engineering team can get to it.  yes, gr8 does start next week.

Reply:
i made the requested changes, thanks @makoto @owocki